src/utils/wallet.ts

```diff
@@ -0,0 +1,220 @@
+  wordsValid(words: string): {valid: boolean, message: string, words?: string} {
+    let newWordsString = '';
+    if (_.isString(words)) {
+      // 1. Replace all non ascii chars by a single space
+      // 2. Remove one or more spaces (or line breaks) before and after the 24 words
+      // 3. Set text to lower case
+      newWordsString = words.replace(/[^A-Za-z0-9]/g, ' ').replace(/\s+/g, ' ').trim().toLowerCase();
+      const wordsArray = newWordsString.split(' ');
+      if (wordsArray.length !== 24) {
+        // Must have 24 words
+        return {'valid': false, 'message': 'Must have 24 words'};
+      } else if (!Mnemonic.isValid(newWordsString)) {
+        // Check if there is a word that does not belong to the list of possible words
+        const wordlist = Mnemonic.Words.ENGLISH;
+        const errorList: string[] = [];
+
+        for (const word of wordsArray) {
+          if (wordlist.indexOf(word) < 0) {
+            errorList.push(word);
+          }
+        }
+
+        let errorMessage = '';
+        if (errorList.length > 0) {
+          errorMessage = `Invalid words: ${errorList.join(' ')}`;
+        } else {
+          // Invalid sequence of words
+          errorMessage = 'Invalid sequence of words';
+        }
+        return {'valid': false, 'message': errorMessage};
+      }
+    } else {
+      // Must be string
+      return {'valid': false, 'message': 'Must be a string'};
+    }
+    return {'valid': true, 'message': '', 'words': newWordsString};
+  },
+
+  /**
+   * Generate HD wallet words
+   *
+   * @param {number} entropy Data to generate the HD Wallet seed - entropy (256 - to generate 24 words)
+   *
+   * @return {string} words generated
+   * @memberof Wallet
+   * @inner
+   */
+  generateWalletWords(entropy: number = HD_WALLET_ENTROPY): string {
+    const code = new Mnemonic(entropy);
+    return code.phrase;
+  },
+
+  /**
+   * Get xpub from data
+   *
+   * @param {Buffer} pubkey Compressed public key
+   * @param {Buffer} chainCode HDPublic key chaincode
+   * @param {Buffer} fingerprint parent fingerprint
+   * @param {string} networkName Optional parameter to select the used network (default is mainnet)
+   *
+   * @return {String} Xpub
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  xpubFromData(pubkey: Buffer, chainCode: Buffer, fingerprint: Buffer, networkName: string = 'mainnet'): string {
+    const network = new Network(networkName);
+    const hdpubkey = new HDPublicKey({
+      network: network.bitcoreNetwork,
+      depth: 4,
+      parentFingerPrint: fingerprint,
+      childIndex: 0,
+      chainCode: chainCode,
+      publicKey: pubkey
+    });
+
+    return hdpubkey.xpubkey;
+  },
+
+  /**
+   * Get compressed public key from uncompressed
+   *
+   * @param {Buffer} pubkey Uncompressed public key
+   *
+   * @return {Buffer} Compressed public key
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  toPubkeyCompressed(pubkey: Buffer): Buffer {
+    const x = pubkey.slice(1, 33);
```

Comment:
Assert the size of `pubKey` before slicing?

Reply:
Done!